File: Go-Solutions/nowcoder/nc117.go
package nowcoder

func mergeTrees117(t1 *TreeNode, t2 *TreeNode) *TreeNode {
	// write code here
	if t1 == nil && t2 == nil {
		return nil
	}

	if t1 == nil && t2 != nil {
		return t2
	}

	if t1 != nil && t2 == nil {
		return t1
	}

	return &TreeNode{Val: t1.Val + t2.Val, Left: mergeTrees117(t1.Left, t2.Left), Right: mergeTrees117(t1.Right, t2.Right)}
}

File: Java-Kotlin/src/main/java/nowcoder/nc117.java
package nowcoder;

import common.datastructure.TreeNode;

public class nc117 {

    public TreeNode mergeTrees(TreeNode t1, TreeNode t2) {
        // write code here

        if (t1 == null && t2 == null) {
            return null;
        }

        if (t1 == null && t2 != null) {
            return t2;
        }

        if (t2 == null && t1 != null) {
            return t1;
        }

        TreeNode node = new TreeNode(t1.val + t2.val);
        node.left = mergeTrees(t1.left, t2.left);
        node.right = mergeTrees(t1.right, t2.right);
        return node;
    }
}
